compiler/src/iree/compiler/Dialect/HAL/IR/HALDialect.cpp

```diff
@@ -142,22 +142,38 @@ class HALAffinityAnalysisDialectInterface
       }
 
       MLIRContext *ctx = getContext();
+      std::optional<IREE::Encoding::UnsupportedEncodingAttr> defaultAttr;
+      auto getDefaultAttr = [&]() {
+        if (defaultAttr) {
+          return defaultAttr.value();
+        }
+        defaultAttr = IREE::Encoding::UnsupportedEncodingAttr::get(ctx);
+        return defaultAttr.value();
+      };
       for (IREE::Stream::AffinityAndOpPair key : batchQueries) {
         auto [affinityAttr, op] = key;
         SetVector<IREE::HAL::ExecutableTargetAttr> resultSet;
         deviceAnalysis.gatherRequiredExecutableTargets(affinityAttr, op,
                                                        resultSet);
         for (auto targetAttr : resultSet) {
-          Attribute result = IREE::Encoding::UnsupportedEncodingAttr::get(ctx);
-          if (auto attr = targetAttr.getConfiguration().getNamed("encoding")) {
-            if (auto encodingLayoutAttr =
-                    dyn_cast<IREE::Encoding::EncodingLayoutAttrInterface>(
-                        attr->getValue())) {
-              result = encodingLayoutAttr.cloneWithSimplifiedConfig(
-                  targetAttr.getConfiguration());
-            }
+          if (!targetAttr.hasConfigurationAttr("encoding")) {
+            layoutAttrs[key].insert(getDefaultAttr());
+            continue;
+          }
+          auto attr = targetAttr.getConfiguration().getNamed("encoding");
+          if (!attr) {
+            layoutAttrs[key].insert(getDefaultAttr());
+            continue;
+          }
```

Comment:
```suggestion
          Attribute attr = targetAttr.getConfigurationAttr("encoding");
          if (!attr) {
            layoutAttrs[key].insert(getDefaultAttr());
            continue;
          }
```
This seems a bit duplicated and I think you can just do the above.

Reply:
Thanks, this is what I'm looking for!

Reply:
oh, the method is available for TargetDeviceAttr, but not ExectuableTargetAttr.

Reply:
I think I do not need the second check because it is already done in first check.